crates/miden-lib/src/errors/tx_kernel_errors.rs

```diff
@@ -149,8 +149,12 @@ pub const ERR_NON_FUNGIBLE_ASSET_PROVIDED_FAUCET_ID_IS_INVALID: MasmError = Masm
 pub const ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_ASSETS_FROM_INCORRECT_CONTEXT: MasmError = MasmError::from_static_str("attempted to access note assets from incorrect context");
 /// Error Message: "attempted to access note inputs from incorrect context"
 pub const ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_INPUTS_FROM_INCORRECT_CONTEXT: MasmError = MasmError::from_static_str("attempted to access note inputs from incorrect context");
-/// Error Message: "attempted to access note sender from incorrect context"
-pub const ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_SENDER_FROM_INCORRECT_CONTEXT: MasmError = MasmError::from_static_str("attempted to access note sender from incorrect context");
+/// Error Message: "attempted to access note metadata from incorrect context"
+pub const ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_METADATA_FROM_INCORRECT_CONTEXT: MasmError = MasmError::from_static_str("attempted to access note metadata from incorrect context");
```

Comment:
I wonder if "incorrect context" isn't too generic to be helpful as a message (applies to all these `_INCORRECT_CONTEXT` errors).

Maybe the message should be something along the lines of:

```
failed to access note metadata of current note because no note is currently being processed
```

What do you think?